server/index.ts

```diff
@@ -46,12 +46,20 @@
 require('source-map-support').install();
 // NOTE: This file intentionally doesn't use too many modern JavaScript
 // features, so that it doesn't crash old versions of Node.js, so we
-// can successfully print the "We require Node.js 18+" message.
+// can successfully print the "We require Node.js 22+" message.
 
 // Check for version
 const nodeVersion = parseInt(process.versions.node);
-if (isNaN(nodeVersion) || nodeVersion < 18) {
-	throw new Error("We require Node.js version 18 or later; you're using " + process.version);
+if (isNaN(nodeVersion) || nodeVersion < 22) {
+	try {
+		// structuredClone was introduced in Node 17, but we'll ask for the most recent "Active LTS" with it to be safe
+		// https://nodejs.org/en/about/previous-releases
+		structuredClone({});
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	} catch (e) {
+		throw new Error("We require Node.js version 22 or later; you're using " + process.version);
+	}
 }
```

Comment:
Man I hate when people remove my useful comments and code.

```suggestion
// I've gotten enough reports by people who don't use the launch
// script that this is worth repeating here
try {
	fetch;
// eslint-disable-next-line @typescript-eslint/no-unused-vars
} catch (e) {
	throw new Error("We require Node.js version 22 or later; you're using " + process.version);
}
```

Reply:
> This is the original comment, from: [0143ae2#diff-e07d531ac040ce3f40e0ce632ac2a059d7cd60f20e61f78268ac3be015b3b28f](https://github.com/smogon/pokemon-showdown/commit/0143ae28dd5a702b4b3eb86eef886535b95b8794#diff-e07d531ac040ce3f40e0ce632ac2a059d7cd60f20e61f78268ac3be015b3b28f)

I like your version more. Feature-testing is better than version number comparison. I'm not exactly opposed to replacing ``structuredClone`` for ``fetch``, but I'm not sure bare ``fetch;`` will do the trick, since there's no ``"use strict";``.

Dependency check at least for ``sockjs`` may be unneeded, since ``sockets`` will error on ``require`` anyway, but no strong opinion.